internal/provider/group_resource.go

```diff
@@ -0,0 +1,345 @@
+	ParseHttpResponse(body []byte) error
+	CreateRequestBody() ([]byte, diag.Diagnostics)
+	GetURL() string
+}
+
+// GroupResource is the resource implementation.
+type GroupResource struct {
+	client ProviderHTTPClient
+}
+
+// Metadata returns the resource type name.
+func (r *GroupResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_group"
+}
+
+// Schema defines the schema for the group resource.
+func (r *GroupResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+
+			"inventory_id": schema.Int64Attribute{
+				Required: true,
+			},
+			"name": schema.StringAttribute{
+				Required: true,
+			},
+			"description": schema.StringAttribute{
+				Optional: true,
+			},
+			"group_url": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"variables": schema.StringAttribute{
+				Optional:   true,
+				CustomType: jsontypes.NormalizedType{},
+			},
+		},
+	}
+}
+
+// GroupResourceModel maps the resource schema data.
+type GroupResourceModel struct {
+	InventoryId types.Int64          `tfsdk:"inventory_id"`
+	Name        types.String         `tfsdk:"name"`
+	Description types.String         `tfsdk:"description"`
+	URL         types.String         `tfsdk:"group_url"`
+	Variables   jsontypes.Normalized `tfsdk:"variables"`
+}
+
+func (d *GroupResourceModel) GetURL() string {
+	if IsValueProvided(d.URL) {
+		return d.URL.ValueString()
+	}
+	return ""
+}
+
+func (d *GroupResourceModel) CreateRequestBody() ([]byte, diag.Diagnostics) {
+	body := make(map[string]interface{})
+	var diags diag.Diagnostics
+
+	// Inventory id
+	body["inventory"] = d.InventoryId.ValueInt64()
+
+	// Name
+	body["name"] = d.Name.ValueString()
+
+	// Variables
+	if IsValueProvided(d.Variables) {
+		body["variables"] = d.Variables.ValueString()
+	}
+
+	// URL
+	if IsValueProvided(d.URL) {
+		body["url"] = d.URL.ValueString()
+	}
+
+	// Description
+	if IsValueProvided(d.Description) {
+		body["description"] = d.Description.ValueString()
+	}
+
+	json_raw, err := json.Marshal(body)
+	if err != nil {
+		diags.Append(diag.NewErrorDiagnostic("Body JSON Marshal Error", err.Error()))
+		return nil, diags
+	}
+	return json_raw, diags
+}
+
+func (d *GroupResourceModel) ParseHttpResponse(body []byte) error {
+	/* Unmarshal the json string */
+	result := make(map[string]interface{})
+
+	err := json.Unmarshal(body, &result)
+	if err != nil {
+		return err
+	}
+
+	d.Name = types.StringValue(result["name"].(string))
+	if result["description"] != "" {
+		d.Description = types.StringValue(result["description"].(string))
+	} else {
+		d.Description = types.StringNull()
+	}
+	d.URL = types.StringValue(result["url"].(string))
+	if result["variables"] != nil {
```

Comment:
I don't think the variables field in the API response will ever be nil, if there are no variables supplied when the group resource is created then the variables field returned will be an empty string so that's probably what we should check for here.